### ISSUE TITLE
Upload multiple image for upscaling?

### ISSUE THREAD

**Issue by tralala87** [lllyasviel/Fooocus#1236]
Would it be possible to modify the code for gradio, so that more than 1 image could be uploaded for upscaling, so to automate upscaling a batch file, instead of uploading one by one?

**Comment by tralala87** [comment]
Anybody? :)

**Comment by mashb1t** [comment]
Yes, this would be possible. Feel free to contribute and create a PR.
You can use https://www.gradio.app/docs/file with file_count = 'multiple' and iterate over the list of files to create an upscaling AsyncJob for each.

**Comment by tralala87** [comment]
I tried in various ways (files and zip), but the uploaded images or zip file are not recognized.

**Comment by mashb1t** [comment]
Please also adjust the single image processing into a foreach loop logic for upscaling. This is currently not implenented, as, you guessed it, only single iamges are processed.

**Comment by mashb1t** [comment]
Please also see https://github.com/lllyasviel/Fooocus/issues/1664 where others are currently working on implementing a prompt queue.

**Comment by mashb1t** [comment]
Ideas from https://github.com/lllyasviel/Fooocus/issues/1853 by @drdancm:

> 1. Since Fooocus makes so many of the operations so much easier than A1111, it would be great to have Fooocus have this autoscaling feature with a user set maximum pixel size for the longest side of an image.
> 2. In addition, would it be possible to also have a batch mode allowing the user to browse for both an input and and output folder.

**Comment by dkutz78** [comment]
I would also like an option to upscale multiple files. How can this be achieved? Can I edit the Gradio UI somehow?

**Comment by mashb1t** [comment]
@dkutz78 see https://github.com/lllyasviel/Fooocus/issues/1236#issuecomment-1872977886, not only UI changes are needed here.

**Comment by dkutz78** [comment]
> @dkutz78 see [#1236 (comment)](https://github.com/lllyasviel/Fooocus/issues/1236#issuecomment-1872977886), not only UI changes are needed here.

Thanks for the quick response. Since I have no coding experience I think it's above my pay grade(io) so to speak. but thanks anyway....

**Comment by MMFJ** [comment]
I have added this feature but I am not experienced with making pull requests and interacting with the repo. If someone wants to add the changes and create a pull request, here is the updated webui.py. For those interested, just backup your existing file and use this one.
[https://gist.github.com/MMFJ/fab678c086761ac2906343337f6c5bb0](https://gist.github.com/MMFJ/fab678c086761ac2906343337f6c5bb0)
